packages/query/src/query-engine/index.ts

```diff
@@ -62,4 +65,95 @@ export class QueryEngine {
 
 		return parseTemplate( data, wpHomeUrl, uri );
 	};
+
+	/**
+	 * Fetches paths to be rendered statically.
+	 *
+	 * @param {GetPagesToRenderStaticallyQueryVariables} variables - The variables for the query.
+	 * @return The path data to be rendered statically.
+	 */
+	static getStaticPaths = async (
+		variables: GetPagesToRenderStaticallyQueryVariables = {}
```

Comment:
Dont bring these types directly into our class, it causes coupling with GraphQL we want to avoid.

Instead our VariableType is validated as compatible with the graphql via `fetchQuery()`